result.go

```diff
@@ -0,0 +1,104 @@
+// Package command provides an organized way of running a *exec.Cmd and generates a Result that contains all
+// commonly-accessed information in the Cmd that allows for repeated execution.
+package command
+
+import (
+	"context"
+	"errors"
+	"os/exec"
+	"path"
+)
+
+// Result represents a single execution of a *exec.Cmd.
+type Result struct {
+	// Name is the command to run, similar to exec.Command's first parameter
+	Name string `json:"name"`
+
+	// Args contains the command line called, without argv[0].
+	Args []string `json:"args,omitempty"`
+
+	// Env contains the environment passed into CaptureContext.
+	Env []string `json:"env,omitempty"`
+
+	// Output provides the combined output from the command as a string.
+	Output string `json:"output,omitempty"`
+
+	// ExitCode holds the exit code returned by the call.
+	// It will be 0 (default value) even if a command didn't run due to error.
+	// You MUST check error when calling any of the functions below, as the
+	// exec.ExitError type contains additional context for failure.
+	ExitCode int `json:"exitCode"`
+}
+
+// CaptureContext executes an exec.CommandContext and returns Result and error.
+//
+// The Result will always be returned, even if it's incomplete.
+//
+// The error type is passed on unwrapped and may contain an *exec.ExitError, which can be converted with errors.As
+// to check for additional information.
+//
+// The parameters behave as they do in *exec.Cmd, so passing in a nil env will inherit the parent environment,
+// and passing an empty slice will create and empty environment.
+func CaptureContext(ctx context.Context, env []string, name string, args ...string) (cr Result, err error) {
+	if ctx == nil {
+		ctx = context.Background()
+	}
+
+	cmd := exec.CommandContext(ctx, name, args...)
+	cmd.Env = env
+
+	return capture(cmd)
+}
+
+// Capture executes an exec.CommandContext and captures the output as a Result and an error. It has no control over
+// the context.Context. If you need to inject context.Context, use CaptureContext.
+func Capture(env []string, name string, args ...string) (cr Result, err error) {
+	return CaptureContext(context.Background(), env, name, args...)
+}
+
+func capture(cmd *exec.Cmd) (cr Result, err error) {
+	output, err := cmd.CombinedOutput()
+
+	cr = Result{
+		Name:     path.Base(cmd.Path),
```

Comment:
when you pass a "name" to `exec.Cmd`, it gets cleaned up, finding it in the path. this version of the code stored the run state in cmd, rather than an outer struct that just holds the original `name` value, regardless of whether it's a fully declared path or not.

in pull #2 we fix this completely by building the structure from the user's point of view, and only add output/exitCode in this function.